shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetReducer.kt

```diff
@@ -214,7 +232,11 @@ class StudyPlanWidgetReducer : StateReducer<State, Message, Action> {
 
     private fun handleActivityClicked(state: State, activityId: Long): StudyPlanWidgetReducerResult {
         val activity = state.activities[activityId]
+
         if (activity?.isCurrent != true) return state to emptySet()
+
+        if (activity.isIdeRequired) return state to setOf(Action.ViewAction.ShowStageImplementUnsupportedModal)
```

Comment:
This check should be performed only for learning activities with the type of `LearningActivityType.IMPLEMENT_STAGE`:
```kotlin
val action = when (activity.type) {
    LearningActivityType.IMPLEMENT_STAGE -> {
        val projectId = state.studyPlan?.projectId
        if (projectId != null && activity.targetType == LearningActivityTargetType.STAGE) {
            if (activity.isIdeRequired) {
                Action.ViewAction.ShowStageImplementUnsupportedModal
            } else {
                Action.ViewAction.NavigateTo.StageImplementation(
                    stageId = activity.targetId,
                    projectId = projectId
                )
            }
        } else {
            null
        }
    }
```